service/view_building_coordinator.cc

```diff
@@ -0,0 +1,513 @@
+    auto filter_building_tasks = [] (const std::vector<view_building_task>& tasks) -> std::vector<utils::UUID> {
+        return tasks | std::views::filter([] (const view_building_task& t) {
+            return t.type == view_building_task::task_type::build_range;
+        }) | std::views::transform([] (const view_building_task& t) {
+            return t.id;
+        }) | std::ranges::to<std::vector>();
+    };
+
+    auto& base_tasks = _vb_sm.building_state.tasks_state[*_vb_sm.building_state.currently_processed_base_table];
+    if (!base_tasks.contains(replica)) {
+        // No tasks for this replica
+        vbc_logger.debug("No task for replica: {}", replica);
+        return {};
+    }
+
+    auto& tablet_map = _db.get_token_metadata().tablets().get_tablet_map(*_vb_sm.building_state.currently_processed_base_table);
+    for (auto& [token, tasks]: collect_tasks_by_last_token(base_tasks[replica])) {
+        auto tid = tablet_map.get_tablet_id(token);
+        if (tablet_map.get_tablet_transition_info(tid)) {
+            vbc_logger.debug("Tablet {} on replica {} is in transition.", tid, replica);
+            continue;
+        }
+
+        auto building_tasks = filter_building_tasks(tasks);
+        if (!building_tasks.empty()) {
+            return building_tasks;
+        } else {
+            return tasks | std::views::transform([] (const view_building_task& t) {
+                return t.id;
+            }) | std::ranges::to<std::vector>();
+        }
+    }
+    vbc_logger.debug("No tasks for replica {} can be started now.", replica);
+    return {};
+}
+
+future<std::vector<mutation>> view_building_coordinator::start_tasks(const group0_guard& guard, std::vector<utils::UUID> tasks) {
+    vbc_logger.info("Starting tasks {}", tasks);
+
+    std::vector<mutation> muts;
+    for (auto& t: tasks) {
+        auto mut = co_await _sys_ks.make_update_view_building_task_state_mutation(guard.write_timestamp(), t, service::view_building::view_building_task::task_state::started);
+        muts.push_back(std::move(mut));
+    }
+    co_return muts;
+}
+
+void view_building_coordinator::attach_to_started_tasks(const locator::tablet_replica& replica, std::vector<utils::UUID> tasks) {
+    vbc_logger.debug("Attaching to started tasks {} on replica {}", tasks, replica);
+    shared_future<> work = work_on_tasks(replica, std::move(tasks));
+    _remote_work.insert({replica, std::move(work)});
+}
+
+future<> view_building_coordinator::work_on_tasks(locator::tablet_replica replica, std::vector<utils::UUID> tasks) {
+    std::vector<view_task_result> results;
+    try {
+        results = co_await ser::view_rpc_verbs::send_work_on_view_building_tasks(&_messaging, replica.host, _as, tasks);
+    } catch (...) {
+        vbc_logger.warn("Work on tasks {} on replica {}, failed with error: {}", tasks, replica, std::current_exception());
+        _vb_sm.event.broadcast();
+        co_return;
+    }
+
+    SCYLLA_ASSERT(tasks.size() == results.size());
```

Comment:
Please change this to `on_internal_error`. `SCYLLA_ASSERT` is too dangerous, it will crash the coordinator node in production, which will cause reelection of another coordinator, then it will most likely crash on the same assertion, and so on.